core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushJoinIntoTableScan.java

```diff
@@ -103,6 +103,13 @@ public boolean isEnabled(Session session)
     @Override
     public Result apply(JoinNode joinNode, Captures captures, Context context)
     {
+        // Avoid running this rule until joins have been reordered and the optimal join distribution type has been
```

Comment:
Does join pushdown really depends on `DetermineJoinDistributionType`? It can only flip sides or change distribution type.

Reply:
It doesn't depend on it - the goal here is to allow join reordering and distribution type to be set before we push it into table scan (because then ReorderJoins or DetermineJoinDistributionType won't run at all).

Reply:
If that are only things it can do, I do not believe `PushJoinIntoTableScan` cares. The logic does not look at those properties and same decision would be made no matter if PJITS is run before or after `DetermineJoinDistributionType`

Reply:
> If that are only things it can do, I do not believe PushJoinIntoTableScan cares

If so, I would add PJITS immediately (possibly with other pushdows) after reorder joins.